ansible/roles/ansible-horizon-plugin/tasks/pre_tasks.yml

```diff
@@ -2,6 +2,27 @@
 #For Debian: apache2
 #For RHEL: httpd
 
+- name: Find Horizon directory path
+  find: 
+    paths: "{{ horizon_virtual_env | dirname }}/"
+    file_type: directory
+    use_regex: yes
+    patterns:
+      - '^.*?horizon*'
```

Comment:
we need to look for the base file name from the user provided path, instead of checking `^.*?horizon*`. 
you can use `{{ horizon_virtual_env | basename }}`.
also make sure, you verify with multiple values such as `/openstack/venvs/horizon-*` , `/openstack/venvs/*`, `/openstack/venvs/horizon-23.1.0`